web-api/src/personalization/rerank.rs

```diff
@@ -69,9 +69,23 @@ pub(super) fn rerank_by_tag_weights(
     rank_keys_by_score(weighted_documents, |w1, w2| w1.cmp(w2).reverse())
 }
 
-pub(super) fn rerank_by_interest_and_tag_weight(
+pub(super) const IGNORE_CURRENT_SCORE: f32 = 1.0;
+
+/// Reranks documents based on a combination of their current score, interest and tag weights.
+///
+/// The `interest_tag_bias` determines the ratio of interest scoring to tag weight scoring.
+///
+/// The `personalization_ratio` determines the ratio of the current score to the combined score of
+/// interest and tag weight.
+///
+/// So the final score/ranking per document is calculated as:
+///
+/// 1. `personalization_score = interest_tag_bias * interest_score + (1-interest_tag_bias) * tag_score`
+/// 2. `personalization_ratio * personalization_score + (1-personalization_ratio) *  current_score`
```

Comment:
🔧 an array of weights is more readable and it also doesn't hide quadratic influences like the current weight formula has, ie we would have a weight array `[interest_weight, tag_weight, elasticsearch_weight]` and the final score would be computed as a weighted sum `interest_weight * interest_score + tag_weight * tag_score + elasticsearch_weight * elasticsearch_score` (renamed `current_` to `elasticsearch_`).

i just discussed with @acrrd regarding the default config and we concluded that we need two different weight arrays for the `semantic_search` and the `personalized_documents` endpoints, we decided for `[0.4, 0.4, 0.2]` and `[0.5, 0.5, 0.0]` as default weights.